zanata-war/src/main/java/org/zanata/adapter/OpenOfficeAdapter.java

```diff
@@ -31,7 +31,7 @@
  */
 public class OpenOfficeAdapter extends OkapiFilterAdapter {
     public OpenOfficeAdapter() {
-        super(prepareFilter(), IdSource.subDocNameAndTextUnitId, true);
```

Comment:
If we really want to avoid this problem, we should be fixing it at the merge algorithm in `ResourceUtils.transferFromTextFlows(List<TextFlow>, HDocument)`. As long as we have the REST API we can never guarantee that id will always be sensible or consistent, so we're better off working well regardless of id, and let clients and adapters use whatever id makes sense for the relevant format or environment.